### PR TITLE
refactor with hooks

### PR DESCRIPTION
**Pull request by hammaazarok**
# **in this milestone I achieved the following:**
✅ refactor to use functional components and hooks
✅ professionally documented my work.
✅ Followed the best practices for JavaScript.
✅ No linter errors.

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -8,59 +8,54 @@ import calculate from '../logic/calculate';
...
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: null,
-      next: null,
-      operation: null,
-    };
-    this.performClick = this.performClick.bind(this);
-  }
+let [obj, setObj] = [{}, () => {}];
```
Comment (zhadier) [inline src/components/calculator.js]: - [x] Kindly remove this declaration as it serves no purpose in your application. You already have a state managing the obj inside your calculator component

File: src/components/calculator.js [hunk 2]
```diff
@@ -8,59 +8,54 @@ import calculate from '../logic/calculate';
...
+  if (next !== null) {
+    screen.innerHTML = next;
+  } else if (total !== null) {
+    screen.innerHTML = total;
+  } else if (next !== null && total !== null) {
+    screen.innerHTML = total;
+  } else {
+    screen.innerHTML = '0';
   }
+};
```
Comment (zhadier) [inline src/components/calculator.js]: - [x] Kindly place this event handler inside your Calculator component/function. For hook-based functions, event handlers for the component are placed inside the main function which is `Calculator` in your case

File: src/components/calculator.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React from 'react';
+import React, { useState } from 'react';
 import calculate from '../logic/calculate';
 /* eslint-disable react/prefer-stateless-function
  ,no-useless-constructor
```
Comment (zhadier) [inline src/components/calculator.js]: - [x] Please remove all of the eslint-disable commands inside your calculator folder. The linters are there for a reason, to ensure that you follow all the best practices. Kindly refrain from using eslint-disable in all of your upcoming projects aswell.

**Review by zhadier — CHANGES_REQUESTED** [review]
# Hi @hammaazarok  👋 

Good job so far! 🌤️ 
There are some issues that you still need to work on to go to the next project but you are almost there! 🍓 



## Highlights
✔️ No linter errors
✔️ GitFlow was set up correctly
✔️ Professional pull request title and description


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question @zhadier so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator.js (hunk 3 above)
Comment (hammaazarok) [reply src/components/calculator.js]: @zhadier 
thanks for your helpful review changes applied.

**Comment by hammaazarok** [comment]
@zhadier 
thanks for your helpful review changes applied.

**Review by keroloussamy — APPROVED** [review]
Hi @hammaazarok, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap

## Optional suggestions
_Check the comments under the review._

- [ ] x

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**